blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/RunSearch.java

```diff
@@ -102,29 +104,16 @@ public static Iterable<BlueRun> findRuns(Job job, final Link parent){
     }
 
     @SuppressWarnings("unchecked")
-    public static Iterable<BlueRun> findRuns(Job job, final Link parent, int start, int limit){
-        final List<BlueRun> runs = new ArrayList<>();
-        Iterable<Job> pipelines;
-        if(job != null){
-            pipelines = Collections.singletonList(job);
-        }else{
-            pipelines = Jenkins.get().getItems(Job.class);
```

Comment:
AFAICT there were no callers actually passing `null` for `job`, so this code was overly complicated; and the reverse sort would not necessarily be correct if you were actually collecting builds from every job in the system (for some reason).